### ISSUE TITLE
[hold on #16660] mWeb - Holding down on the modal buttons selects words on mWeb vs does nothing on android

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#23736]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to settings > workspaces > select any workspace which doesn't have its default currency set to USD > bank account
2. Hold down on one of the buttons inside the modal
3. Do step 1 and 2 on android and compare the results

## Expected Result:
Both should have consistent behaviors, mWeb should not select the last word.

## Actual Result:
While holding down on the buttons of the modal the last character is selected on mWeb

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [x] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3-45.7
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/318b39bc-aeec-4307-a085-60d2ea7892fb

https://github.com/Expensify/App/assets/93399543/8ba7f539-4bf0-4ed7-9a78-41687ed07fa0

**Expensify/Expensify Issue URL:**
**Issue reported by:** @Nathan-Mulugeta
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1690389791471579

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by ygshbht** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
On mobile web, holding down on modal buttons leads to unintended text selection, which is not the case in the Android app. This inconsistency detracts from the uniformity of the user experience across different platforms.

### What is the root cause of that problem?
The current implementation does not include a functionality to disable text selection when a button is pressed on mobile web.

### What changes do you think we should make in order to solve the problem?
To create consistent behaviour across platforms, we should introduce functionality to block text selection when a button is pressed. This could be achieved by leveraging `ControlSelection.block `on `onPressIn` and `ControlSelection.unblock `on `onPressOut` for the buttons. Alternatively, we could set the CSS property `user-select` to `none` for text we don't want to select.

**Comment by joekaufmanexpensify** [comment]
I can reproduce. Clarifying what part of this is a bug in the thread.

**Comment by Nathan-Mulugeta** [comment]
I want to point out that there was a similar issue [here](https://github.com/Expensify/App/issues/22509) as well and that was fixed.

**Comment by ygshbht** [comment]
Because when a user is pressing a button he is not looking to select?

**Comment by joekaufmanexpensify** [comment]
@parasharrajat curious if you think this one will be fixed by the new version of RN-web? I saw you advised that we close a similar issue for the same reason [here. ](https://github.com/Expensify/App/issues/18553#issuecomment-1539399850)

**Comment by joekaufmanexpensify** [comment]
Put on hold for https://github.com/Expensify/App/issues/16660 !

**Comment by joekaufmanexpensify** [comment]
Still on hold

**Comment by joekaufmanexpensify** [comment]
Same

**Comment by joekaufmanexpensify** [comment]
Still on hold

**Comment by joekaufmanexpensify** [comment]
Held

**Comment by joekaufmanexpensify** [comment]
Same

**Comment by joekaufmanexpensify** [comment]
Still held

**Comment by joekaufmanexpensify** [comment]
Same

**Comment by joekaufmanexpensify** [comment]
Same

**Comment by joekaufmanexpensify** [comment]
Held

**Comment by joekaufmanexpensify** [comment]
Same

**Comment by joekaufmanexpensify** [comment]
Held

**Comment by joekaufmanexpensify** [comment]
Same

**Comment by joekaufmanexpensify** [comment]
Same.

**Comment by joekaufmanexpensify** [comment]
same

**Comment by joekaufmanexpensify** [comment]
Same.

**Comment by joekaufmanexpensify** [comment]
Same

**Comment by joekaufmanexpensify** [comment]
Held

**Comment by joekaufmanexpensify** [comment]
Confirmed this is fixed. Closing as no action is needed in this issue!

**Comment by Nathan-Mulugeta** [comment]
Hey @joekaufmanexpensify reporting bonus has not been issued.

**Comment by joekaufmanexpensify** [comment]
@Nathan-Mulugeta I don't think this is eligible for a reporting bonus. This bug was fixed by https://github.com/Expensify/App/issues/16660 , which was opened prior to this bug being reported in this issue. 

Let me know if you have any questions about this.

**Comment by Nathan-Mulugeta** [comment]
I noticed that in the past, when long standing issues were resolved, reporting bonuses were typically issued as a way to acknowledge the effort. I was simply following that precedent. However, I trust your judgment to handle this situation as you see fit. @joekaufmanexpensify

**Comment by Nathan-Mulugeta** [comment]
Here's an example: [Issue #21453](https://github.com/Expensify/App/issues/21453) showcases a similar scenario where addressing long standing issues was rewarded with bonuses. I hope this helps illustrate the context.

**Comment by joekaufmanexpensify** [comment]
Got it. I can't speak for other situations where we might have issued one-off reporting bonuses. But my understanding is our general policy is that the reporting bonus (if one is applicable) is due for the earliest issue opened that fixed a bug. 

In this case, that would be the other issue this one was held on. My thinking is we should follow that general guidance here, because that issue would have fixed this bug, even if this issue had never been opened. 

We appreciate you reporting this. Please let me know if you have any other questions!

**Comment by Nathan-Mulugeta** [comment]
Its super clear, thanks a lot.
